packages/x-charts/src/internals/plugins/corePlugins/useChartDimensions/useChartDimensions.ts

```diff
@@ -3,13 +3,71 @@
 import * as React from 'react';
 import useEnhancedEffect from '@mui/utils/useEnhancedEffect';
 import ownerWindow from '@mui/utils/ownerWindow';
-import { DEFAULT_MARGINS } from '../../../../constants';
+import { DEFAULT_MARGINS, EMPTY_SIDES } from '../../../../constants';
 import { ChartPlugin } from '../../models';
-import { UseChartDimensionsSignature } from './useChartDimensions.types';
+import type { UseChartDimensionsSignature } from './useChartDimensions.types';
 import { selectorChartDimensionsState } from './useChartDimensions.selectors';
+import { defaultizeMargin } from '../../../defaultizeMargin';
+import { useSelector } from '../../../store/useSelector';
+// TODO: fix dep cycle, should we move selectors to their own folder/file? Eg:
+// useChartCartesianAxis/selectors/chartXAxis.ts
+import {
+  selectorChartXAxis,
+  selectorChartYAxis,
+} from '../../featurePlugins/useChartCartesianAxis/useChartCartesianAxis.selectors';
+import { createSelector } from '../../utils/selectors';
+import { isBandScale } from '../../../isBandScale';
+import { isInfinity } from '../../../isInfinity';
 
 const MAX_COMPUTE_RUN = 10;
 
+const selectorChartAxisSize = createSelector(
+  [selectorChartXAxis, selectorChartYAxis],
+  (xAxis, yAxis) => {
+    const topBottom = xAxis.axisIds.reduce(
+      (acc, id) => {
+        const axis = xAxis.axis[id];
+
+        // TODO: move this to the selectorChartXAxis, axis.isUsed? axis.shouldRender?
```

Comment:
@mui/xcharts what do you think about moving this logic as a permanent part of the axis definition? This can then be used in the x/y axis renderers

Reply:
> I thaugth it's the role of `position?: 'left'|'right` with something like
> 
> ```js
> axis.isUsed = !axis.position===undefined
> ```

Not really, because of 

```
  // Skip axis rendering if no data is available
  // - The domain is an empty array for band/point scales.
  // - The domains contains Infinity for continuous scales.
  if ((ordinalAxis && domain.length === 0) || (!ordinalAxis && domain.some(isInfinity))) {
    return null;
  }
```

We could remove it, but it would only display a line, and we would have to fix some other issues with having bad values 🤔

Reply:
Those are pathological cases where there is no data to display. The axis should exist and we should reserve some space for it when data gets loaded.

Here is an illustration where I modify the left margin when there is no data for the y-axis

https://codesandbox.io/p/sandbox/pensive-dust-g5jl44?file=%2Fsrc%2FDemo.tsx%3A15%2C51

https://github.com/user-attachments/assets/212c8251-3d77-4304-a808-24586cf45c0a

Reply:
Got it, makes sense